crates/ibc/src/core/timestamp.rs

```diff
@@ -75,12 +75,10 @@ impl scale_info::TypeInfo for Timestamp {
     fn type_info() -> scale_info::Type {
         scale_info::Type::builder()
             .path(scale_info::Path::new("Timestamp", module_path!()))
-            // i128 is chosen before we represent the timestamp is nanoseconds, which is represented as a i128 by Time
-            .composite(scale_info::build::Fields::named().field(|f| {
-                f.ty::<Option<i128>>()
-                    .name("time")
-                    .type_name("Option<i128>")
-            }))
+            .composite(
+                scale_info::build::Fields::named()
+                    .field(|f| f.ty::<u64>().name("time").type_name("u64")),
```

Comment:
From what I see, `Timestamp` exceeds 8 bytes, and the `scale_info::Type` with a `time: u64`, will not faithfully represent the `Timestamp`. Wouldn't be an issue with overflow somewhere later?

Reply:
Maybe force conversion to u128？

Reply:
I believe using `u64` here is appropriate because I am using the [`nanoseconds()`](https://github.com/cosmos/ibc-rs/blob/d59b303e09b3fbb415d89466dee5e9a119779aa8/crates/ibc/src/core/timestamp.rs#L56) method of the `Timestamp` struct, which is encoded as a `u64` in the `parity-scale-codec crate`. This replaces the original `Option<Time>` field in the `Timestamp` struct with a `u64`. Therefore, I think it is reasonable to declare `Timestamp` as a `u64` here, and when restoring it, I am using the `Timestamp::from_nanoseconds` method to restore it. I believe using `u64` to represent the encoding of `Timestamp` after being encoded by `parity-scale-codec` is appropriate.

Reply:
> This replaces the original `Option<Time>` field in the Timestamp struct with a u64

:+1: